src/mono/mono/mini/debugger-agent.c

```diff
@@ -1343,32 +1106,39 @@ socket_transport_connect (const char *address)
 	} else {
 		/* Connect to the specified address */
 		/* FIXME: Respect the timeout */
-		for (rp = result->entries; rp != NULL; rp = rp->next) {
-			MonoSocketAddress sockaddr;
-			socklen_t sock_len;
+		uint32_t startTime = time(NULL);
```

Comment:
A less racy way of implementing this is having the other side connect, i.e. open a socket, wait on it, then spawn the debuggee process, and instruct it to connect to the the socket.

Reply:
This will work on mobile devices?

Reply:
Oh, I think I understood your suggestion, but as I understood coreclr first starts the debugee process and after it loads mscordbi, in our case where we will open the socket. If we follow the same order that they do we will need this loop.